service/cluster/psmdb_cluster.go

```diff
@@ -129,18 +129,29 @@ func (s *PSMDBClusterService) UpdatePSMDBCluster(ctx context.Context, req *contr
 
 	params := &k8sclient.PSMDBParams{
 		Name: req.Name,
-		Size: req.Params.ClusterSize,
-		Replicaset: &k8sclient.Replicaset{
-			ComputeResources: new(k8sclient.ComputeResources), // this must be present for a valid request
-		},
 	}
 
-	if req.Params.Replicaset.ComputeResources.CpuM > 0 {
-		params.Replicaset.ComputeResources.CPUM = convertors.MilliCPUToStr(req.Params.Replicaset.ComputeResources.CpuM)
-	}
+	if req.Params != nil {
+		if req.Params.Suspend && req.Params.Resume {
+			return nil, status.Error(codes.InvalidArgument, "field suspend and resume cannot be true simultaneously")
+		}
 
-	if req.Params.Replicaset.ComputeResources.MemoryBytes > 0 {
-		params.Replicaset.ComputeResources.MemoryBytes = convertors.BytesToStr(req.Params.Replicaset.ComputeResources.MemoryBytes)
+		params.Suspend = req.Params.Suspend
+		params.Resume = req.Params.Resume
+		params.Size = req.Params.ClusterSize
+
+		if req.Params.Replicaset != nil {
+			params.Replicaset = new(k8sclient.Replicaset)
+			if req.Params.Replicaset.ComputeResources != nil {
```

Comment:
Can we extract this logic to separate function?

Reply:
Reused `computeResources` func